### ISSUE TITLE
HTTP resource (Resource Adapters)

### ISSUE THREAD

**Issue by adrianthedev** [avo-hq/avo#1418]
### Feature

Right now Avo supports ActiveRecord, but we'd love to have adapters for other types of connections (`mongo`, `http`, `PORO`s)

I did an exploration into this in https://github.com/avo-hq/avo-3/pull/4

### Must have's:

 - [ ] support https://github.com/rails/activeresource
 - [ ] support custom HTTP endpoints with the option to customize where to fetch the data, page, per_page, etc. from the payload (maybe even headers)
 - [ ] support all CRUD operations

### Nice to have's

 - [ ] support array models
 - [ ] filters (server side/API level)
 - [ ] actions
 - [ ] sorting
 - [ ] mongodb

### Other resources

[rails/activeresource](https://github.com/rails/activeresource)
[calebporzio/sushi](https://github.com/calebporzio/sushi)

https://github.com/makandra/active_type

**Comment by elassadi** [comment]
hi Adrian , 
We should have an open discussion about the approach if you need a sparing partner :) 
I was thinking about the same, since not all our services are persisted in traditional Database, we are limited into the usage of avo as the backend tool of choice. 
To make the developer experience simpler it would be awesome if avo supported a more generic data handling with 
multiple adapter

I think the biggest problem here is that Avo is relaying on AR model reflection.
This is something for the roadmap with clear time definition and scope. ? wdyt

**Comment by adrianthedev** [comment]
Yes. Definitely, I'd love to hear your thoughts and do some pairing together. I'll contact you on Discord to set something up.

**Comment by adrianthedev** [comment]
Regarding the roadmap, I definitely want to have this in for Avo v3.

**Comment by Paul-Bob** [comment]
Closed by https://github.com/avo-hq/avo-http_resource/pull/1
